R/workflows.R

```diff
@@ -13,44 +13,34 @@
 ##' @return List with key report name and value is any missing dependencies.
 ##'
 ##' @keywords internal
-workflow_missing_dependencies <- function(path, reports) {
-  all_reports <- orderly::orderly_list(root = path)
-  report_names <- lapply(reports, function(report) report$name)
-  missing_deps <- lapply(reports, function(report) {
-    if (!(report$name %in% all_reports)) {
-      stop(sprintf("Report with name '%s' cannot be found.", report$name))
+workflow_missing_dependencies <- function(path, reports, ref = NULL) {
+  report_names <- vcapply(reports, function(report) report$name,
+                          USE.NAMES = FALSE)
+  dependencies <- orderly_upstream_dependencies(report_names, root = path,
+                                                ref = ref)
+  missing_deps <- lapply(report_names, function(report) {
+    report_deps <- dependencies[[report]]
+    if (is.null(report_deps)) {
+      return(list())
     }
-    get_missing_dependencies(report$name, path, report_names)
+    recursive_scalar(
```

Comment:
At some point it might be worth thinking about how much serialisation infects the logic places like this